src/DiseaseStatus.H

```diff
@@ -217,8 +201,8 @@ void DiseaseStatus<AC,ACT,ACTD,A>::updateAgents(AC& a_agents, /*!< Agent contain
             ParallelFor( np, [=] AMREX_GPU_DEVICE (int i) noexcept
             {
                 if (flag_hosp_ptr[i] == 1) {
-                    AMREX_ALWAYS_ASSERT(!isHospitalized(i, ptd));
-                    assign_hospital( i, hosp_i_ptr, hosp_j_ptr, ptd);
+                    // could have already been assignedto a hospital from another disease
+                    if (!isHospitalized(i, ptd)) assign_hospital( i, hosp_i_ptr, hosp_j_ptr, ptd);
```

Comment:
I don't think the agent has already been assigned to a hospital. We are now outside the `for (int d = 0; d < n_disease; d++ ) {` loop. After it iterates through all diseases, if the `flag_hosp_ptr[i]` is set to `1` (i.e., marked for hospitalization for at least one disease), the agent gets assigned a hospital.